crates/ruff_python_formatter/src/expression/expr_bin_op.rs

```diff
@@ -36,68 +36,137 @@ impl FormatNodeRule<ExprBinOp> for FormatExprBinOp {
+                // let right = right.format().memoized();
+                // write!(
+                //     f,
+                //     [best_fitting![
+                //         format_args![left, space(), op.format(), space(), right],
+                //         format_args![group(&left), space(), op.format(), space(), right],
+                //         format_args![
+                //             text("("),
+                //             block_indent(&format_args![
+                //                 left,
+                //                 hard_line_break(),
+                //                 op.format(),
+                //                 space(),
+                //                 right
+                //             ]),
+                //             text(")")
+                //         ]
+                //     ]]
+                // )
+            }
+
+            BinaryLayout::ExpandRight => {
+                let left_group = f.group_id("BinaryLeft");
+
+                write!(
+                    f,
+                    [
+                        // Wrap the left in a group and gives it an id. The printer first breaks the
+                        // right side if `right` contains any line break because the printer breaks
+                        // sequences of groups from right to left.
+                        // Indents the left side if the group breaks.
+                        group(&format_args![
+                            if_group_breaks(&text("(")),
+                            indent_if_group_breaks(
+                                &format_args![
+                                    soft_line_break(),
+                                    left.format(),
+                                    soft_line_break_or_space(),
+                                    op.format(),
+                                    space()
+                                ],
+                                left_group
+                            )
+                        ])
+                        .with_group_id(Some(left_group)),
+                        // Wrap the right in a group and indents its content but only if the left side breaks
+                        group(&indent_if_group_breaks(&right.format(), left_group)),
+                        // If the left side breaks, insert a hard line break to finish the indent and close the open paren.
+                        if_group_breaks(&format_args![hard_line_break(), text(")")])
+                            .with_group_id(Some(left_group))
+                    ]
+                )
             }
 
-            write!(f, [group(&right.format())])
+            BinaryLayout::ExpandRightThenLeft => {
+                write!(
+                    f,
+                    [
+                        group(&left.format()),
+                        space(),
+                        op.format(),
+                        space(),
+                        group(&right.format())
```

Comment:
The formatter breaks group-sequences from right to left, so that works just fine

Reply:
please add as a source code comment